### PR TITLE
Add tests for invalid config of github and office365 modules

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/4983|

## Description

This PR includes:

- Tests for `github` module configuration.
- Tests for `office365` module configuration.

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.
- [x] `provision_documentation.sh` generate the docs without

### CODE REVIEW

**Comment by spothound** [comment]
I'm reviewing this.

File: tests/integration/test_office365/test_configuration/test_invalid.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                                callback=callback_detect_subscription_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    else:
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_read_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'wm_gcp_read(): ERROR:').result()
```
Comment (spothound) [inline tests/integration/test_office365/test_configuration/test_invalid.py:247]: All this code could be merged if you define a map where the keys are the content of tags_to_apply and the values are the callbacks, then you would just do:

```
wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
                                callback=mymap[tags_to_apply],
                                accum_results=1,
                                error_message='Did not receive expected '
                                              'wm_gcp_read(): ERROR:').result()
```

I think we could do this to make the code easier to read and maintain, but it works perfectly like it is, so this change is up to you!

File: tests/integration/test_github/test_configuration/test_invalid.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                                callback=callback_detect_event_type_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'sched_scan_validate_parameters(): ERROR').result()
+    else:
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_read_err,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                              'wm_gcp_read(): ERROR:').result()
```
Comment (spothound) [inline tests/integration/test_github/test_configuration/test_invalid.py:247]: This could probably be merged using a map.

**Review by spothound — COMMENTED** [review]
Ok, @TomasTurina, the code seems perfect to me, I've just noticed some unused import that I would like to remove. Also, I see some duplicated code that we may merge to make it more maintainable and readable, just a suggestion.

Finally, I have detected that those tests require the `wazuh_modules.debug=2` value set in internal options to work. We **are deprecating the option to add internal options on Jenkins**, currently, we're encouraging the team to use a fixture to automatically set the required internal options before running the tests, so **I would like to ask you to add this fixture to remove the requirement of setting this option previous to the test.**

You have an example on how to do this here: https://github.com/wazuh/wazuh-qa/blob/e5d60203098b39128acb8d32f45571199d0e1a9e/tests/integration/test_logcollector/test_location_custom_sockets/test_location_custom_sockets.py#L32

image

Please have all of this in consideration and ask us for review again when you finish making changes to the branch.

Best regards.

File: deps/wazuh_testing/wazuh_testing/github.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply deps/wazuh_testing/wazuh_testing/github.py]: Done!

File: deps/wazuh_testing/wazuh_testing/office365.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply deps/wazuh_testing/wazuh_testing/office365.py]: Done!

File: tests/integration/test_office365/test_configuration/test_invalid.py (hunk 1 above)
Comment (TomasTurina) [reply tests/integration/test_office365/test_configuration/test_invalid.py]: Done!

File: tests/integration/test_github/test_configuration/test_invalid.py (hunk 2 above)
Comment (TomasTurina) [reply tests/integration/test_github/test_configuration/test_invalid.py]: Done!

**Comment by spothound** [comment]
The requested changes were applied by TomasTaurina and now I am going to launch automatic tests again to confirm that now everything (including the automatic internal option configuration) works as expected.


|Wazuh branch| Tested commit | Revision |
|---| --- | --- |
|dev-office365-module| d5925fc81c32829b5f153b1e020386e14a1c8ac9 | 0.commitd5925fc |

## Testing
### `test_github`

#### Centos


### Ubuntu


### `test_office`

#### Centos


#### Ubuntu

**Review by spothound — APPROVED** [review]
Awesome job, thank you very much!

**Pull request merged**